src/simulators/ch/ch_runner.hpp

```diff
@@ -0,0 +1,661 @@
+#ifndef chsimulator_runner_hpp
+#define chsimulator_runner_hpp
+
+#include "chlib/core.hpp"
+#include "chlib/chstabilizer.hpp"
+#include "gates.hpp"
+
+#include "framework/json.hpp"
+#include "framework/operations.hpp"
+#include "framework/rng.hpp"
+#include "framework/types.hpp"
+
+
+#define _USE_MATH_DEFINES
+#include <math.h>
+
+#include <cstdint>
+#include <complex>
+#include <vector>
+
+#ifdef _OPENMP
+#include <omp.h>
+#endif
+
+namespace CHSimulator {
+  
+using complex_t = std::complex<double>;
+using uint_t = uint_fast64_t;
+using int_t = int_fast64_t;
+using chstabilizer_t = StabilizerState;
+
+double t_angle = M_PI/4.;
+double tdg_angle = -1.*M_PI/4.;
+
+const U1Sample t_sample(t_angle);
+const U1Sample tdg_sample(tdg_angle);
+
+uint_t zero = 0ULL;
+
+class Runner
+{
+private:
+  uint_t n_qubits;
+  uint_t chi;
+  uint_t three_qubit_gate_count; //Needed for normalising states
+  std::vector<chstabilizer_t> states;
+  std::vector<complex_t> coefficients;
+  uint_t n_omp_threads;
+  uint_t omp_threshold;
+
+  bool accept;
+  complex_t old_ampsum;
+  uint_t x_string;
+  uint_t last_proposal;
+
+  std::unordered_map<double, U1Sample> z_rotations;
+
+  void InitMetropolis(AER::RngEngine &rng);
```

Comment:
We always use  `lower_snake_case` for naming functions and variables, and `UpperCamelCase` for classes.
so this should be: 
```c++
void init_metropolis( ... )
```